Kernel/Storage/NVMe/NVMeNameSpace.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021, Pankaj R <pankydev8@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "NVMeNameSpace.h"
+#include <AK/NonnullOwnPtr.h>
+#include <Kernel/Devices/DeviceManagement.h>
+#include <Kernel/Storage/NVMe/NVMeNameSpace.h>
+#include <Kernel/Storage/StorageManagement.h>
+
+namespace Kernel {
+
+ErrorOr<NonnullRefPtr<NVMeNameSpace>> NVMeNameSpace::try_create(NonnullRefPtrVector<NVMeQueue> queues, u8 controller_id, u16 nsid, size_t storage_size, size_t lba_size)
+{
+    auto minor_number = StorageManagement::minor_number();
+    auto major_number = StorageManagement::major_number();
+    auto device_name = String::formatted("nvme{:d}n{:d}", controller_id, nsid);
+    auto device_name_kstring = KString::must_create(device_name.view());
+    auto device = TRY(DeviceManagement::try_create_device<NVMeNameSpace>(queues, storage_size, lba_size, major_number, minor_number, nsid, move(device_name_kstring)));
+
+    return device;
+}
+
+NVMeNameSpace::NVMeNameSpace(NonnullRefPtrVector<NVMeQueue> queues, size_t max_addresable_block, size_t lba_size, size_t major_number, size_t minor_number, u16 nsid, NonnullOwnPtr<KString> dev_name)
+    : StorageDevice(major_number, minor_number, lba_size, max_addresable_block, move(dev_name))
+    , m_nsid(nsid)
+    , m_lba_size(lba_size)
+    , m_max_addresable_block(max_addresable_block)
```

Comment:
Do we actually need this? StorageDevice already has such member.

Reply:
Good point! Also removed lba_size because block device has that member!